spyder_kernels/comms/__init__.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2009- Spyder Kernels Contributors
+#
+# Licensed under the terms of the MIT License
+# (see spyder_kernels/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""
+API to communicate between the spyder IDE and the spyder kernel.
+It uses IPython Comms for messaging. The messages are sent by calling an
+arbitrary function, with the limitation that the arguments have to be
+picklable. If the function must return, the call must be blocking.
+
+In addition, the frontend can interrupt the kernel to process the message sent.
```

Comment:
Well this is a bit hard. Basically the kernel will get messages, but it is not checking them while the execution takes place. Therefore the kernel needs to be interrupted to process the messages. This is because coroutines are used instead of threads in ipykernel.
While debugging, this is done by sending a command to pdb:
`u"!get_ipython().kernel.frontend_comm.remote_call(blocking=True).pong()"
For general execution, the solution I found was to use signals (https://github.com/spyder-ide/spyder-kernels/pull/117), but this only work in unix. Another solution must be found for windows (or any platform that doesn't have `SIGUSR1`).

Reply:
Ok, thanks for the explanation. It's not really critical to have that functionality yet.